fs/fcb/src/fcb_area_info.c

```diff
@@ -21,24 +21,28 @@
 #include "fcb_priv.h"
 
 int
-fcb_area_info(struct fcb *fcb, struct flash_area *fa, int *elemsp, int *bytesp)
+fcb_area_info(struct fcb *fcb, int sector, int *elemsp, int *bytesp)
 {
     struct fcb_entry loc;
+    struct fcb_sector_info info;
     int rc;
     int elems = 0;
     int bytes = 0;
 
-    loc.fe_area = fa;
+    fcb_get_sector_info(fcb, sector, &info);
+    loc.fe_range = info.si_range;
+    loc.fe_sector = info.si_sector_in_range + loc.fe_range->sr_first_sector;
     loc.fe_elem_off = 0;
+    /* In case caller passed oldest, get real sector number */
+    if (sector == FCB_SECTOR_OLDEST)
```

Comment:
I believe there is as coding standard requirement to use `{` / `}` for all `if` statements.